### PR TITLE
Implement initialization + tracking of kubectl proxy process

### PR DESCRIPTION
**Pull request by jinnovation**
Contributes to #235.

This PR introduces the `kubernetes--with-proxy` macro, which allows for
executing a block of code with the guarantee of a `kubectl proxy` kicking off,
or having already kicked off, in the background. This paves the way for us to
make HTTP requests to the Kubernetes API via the proxy.

One downside here is that the proxy process lives indefinitely once it's kicked
off. Ideally, this process would have a TTL where it shuts down after going
unused for a specified amount of time. This can be future work. Due to this,
however, we should only merge this PR in once #232 is addressed.

### CODE REVIEW

File: kubernetes-process.el
Comment (jinnovation) [reply kubernetes-process.el:105]: Done.

File: kubernetes.el
Comment (jinnovation) [inline kubernetes.el:41]: Given that the process ledger doesn't have a TTL mechanism for the proxy process, let's compromise and give the `--with-proxy` macro the option to explicitly request to terminate the proxy after it's done. 
```emacs-lisp
(defmacro kubernetes--with-proxy (&optional terminate-on-done &rest body)
  ...)
```

This would require `get-proxy-process` to return a sentinel value denoting whether a new proxy process was created, or if it's simply returning an existing one. If the latter, then `terminate-on-done` shall be a no-op.

Naturally, this will also require a `release-proxy-process` method on the process ledger. This can be a no-op if there's no existing proxy process.

We might also consider having `--with-proxy` kick off a timer, after which `release-proxy-process` gets called regardless.

Comment (jinnovation) [reply kubernetes.el:41]: I mostly just think it'd be a terrible idea to have a live proxy running hidden in the background indefinitely.

File: kubernetes-process.el
Comment (jinnovation) [inline kubernetes-process.el:100]: Consider using the standard `kubernetes-kubectl` function here.

**Review by noorul — APPROVED** [review]

**Comment by noorul** [comment]
I will be offline till Jan 7th

**Comment by jinnovation** [comment]
Going to hold off on moving forward w/ this PR until #247 is addressed.

**Comment by jinnovation** [comment]
Going to close this PR as it's been made largely redundant by #252. Will create a new one targeting specifically the `--with-proxy` macro once #252 is approved and merged.

**Pull request closed without merging**
